Subject: [PATCH] refpolicy: fix policy for "getty console"

When "getty console" adding to inittab, SELinux will block users
login in, because no allow rules for locallogin_t to ioctl on
console_device_t. So, fix it.

This is the login process.
 > /sbin/[min]getty console (getty_t)
    main()
    `- open_tty (); // open tty device as stdin/stdout/stderr,
                    // current tty device is *console_device_t*
    `- execl("/bin/login", ...);
 > /bin/login -- username (locallogin_t)
    main()
    `- setup_tty()
       `- tcsetattr/ioctl(0, ..) // We are blocked here!!
    `- pam session process
       `- pam_selinux.so // relabel dev_console_t to user_tty_device_t

Upstream-Status: Pending

Signed-off-by: Xin Ouyang <Xin.Ouyang@windriver.com>
---
 policy/modules/system/locallogin.te |    1 +
 1 file changed, 1 insertion(+)

--- refpolicy.orig/policy/modules/system/locallogin.te
+++ refpolicy/policy/modules/system/locallogin.te
@@ -152,6 +152,7 @@ ifdef(`distro_ubuntu',`
 tunable_policy(`console_login',`
 	# Able to relabel /dev/console to user tty types.
 	term_relabel_console(local_login_t)
+	term_read_console(local_login_t)
 ')
 
 tunable_policy(`use_nfs_home_dirs',`
